src/test/ui/block-result/block-must-not-have-result-while.stderr

```diff
@@ -0,0 +1,14 @@
+error[E0308]: mismatched types
+  --> $DIR/block-must-not-have-result-while.rs:13:9
+   |
+11 | fn main() {
+   |           - expected `()` because of this default return type
+12 |     while true {
```

Comment:
The code now [walks up](https://github.com/rust-lang/rust/pull/42850/files#diff-2611048bb760f028c808ed4a14b07c00R635) until it finds a `fn` block for success or a `while` or `for` block for failure, avoiding this.